docs/docsite/rst/reference_appendices/glossary.rst

```diff
@@ -475,6 +475,32 @@ when a term comes up on the mailing list.
         inferred from the remote system's current state or nature (which is
         what Facts are).
 
+    Variable Expansion
+        In general, Ansible expands any variables in playbook content at the last
+        possible second, which means that if you define a data structure that data
+        structure itself can define variable values within it, and everything "just
+        works" as you would expect. This also means variable strings can include other
+        variables inside of those strings. This is known as Lazy Evaluation (defined
+        above), which allows for variables to be defined out of order so long as they
+        are defined before usage.
+
+        ::
+
+            vars:
+                a: "{{ b }}"
+                b: 3
+
+        When using an Ansible :ref:`Lookup Plugin`, do note that the value will be
+        evaluated every time the variable is expanded in a template or in a playbook
+        unless it is explicitly used with ``set_fact`` which will evaluate and define
```

Comment:
dont put set_fact here, explain below that how it works as it is a task option

Reply:
I'm sorry but I don't follow ... what do you mean "how it works as it is a task option"?

Reply:
friendly nag: ping @bcoca

Reply:
that 'set_fact' is not magic onto itself, the reason it works is because module options are evaluated